### ISSUE TITLE
Modifying a Plotly figure makes figure layout inconsistent when using toolbar

### ISSUE THREAD

**Issue by julioasotodv** [holoviz/panel#1801]
Hi,

I have been trying `pn.pane.Plotly` lately, and I have came upon a weird behaviour I would like to share with you.

It turns out that modifying the elements plotted in a Plotly figure (this is, the figure traces) is straightworfard and can be done through callbacks, which is great.

However, once a figure has been modified, Plotly's tools (zoom, rotation in the case of 3D figures, select, etc) reset the viewport set by the user through this tools (called in Panel as the `relayout_data` attribute for Plotly panes).

Here's a minimal example (tested with Panel 0.10.1 and Plotly 4.12.0) that can be ran in a notebook. In this case, I create a Plotly pane and a button that modifies the plotted data, which works perfectly. But once the button is clicked, the viewing position is messed up every time I click on a different tool:

```python
import numpy as np
import pandas as pd
import panel as pn
import plotly.graph_objs as go

pn.extension("plotly")

# Generate some data for Plotly 3D chart:
xx = np.linspace(-3.5, 3.5, 100)
yy = np.linspace(-3.5, 3.5, 100)
x, y = np.meshgrid(xx, yy)
z = np.exp(-(x-1)**2-y**2)-(x**3+y**4-x/5)*np.exp(-(x**2+y**2))

# Plotly figure:
surface = go.Surface(z=z)
layout = go.Layout(
    title='Plotly 3D Plot',
    autosize=False,
    width=500,
    height=500,
    margin=dict(t=50, b=50, r=50, l=50),
    uirevision=True # Lets Plotly to keep layout and view angle even after data updates in the figure
)

# Create Plotly pane:
fig = dict(data=[surface], layout=layout)
plotly_pane = pn.pane.Plotly(fig)

# Button that squares the Z coordinates:
a_button = pn.widgets.Button(name="Square z coordinates!")

# Callback for button: modifies underlying data for Plotly figure 
# and updates the pane:
def modify_plotly_figure(event):
    surface["z"] = surface["z"] ** 2
    plotly_pane.object = fig
    
a_button.on_click(modify_plotly_figure)

# Display both elements in a column:
pn.Column(a_button, plotly_pane)
```

The behaviour looks as follows:

![plotly_panel](https://user-images.githubusercontent.com/20630819/100011510-21e57500-2dd2-11eb-817a-d2d40f9810ca.gif)

So, everything looks great until I click the button (tools work as expected). Once clicked, the figure is modified correctly, but from that point the viewing angle is messed up every time I switch tools in the toolbar.

A hacky workaround would be to unlink Panel's JS watcher for the `relayout_data`, but I am having a hard time trying to remove callbacks that come with a specific pane (I don't even know if it is allowed).

Any suggestions are more than welcome.

Thank you for the great work!

**Comment by julioasotodv** [comment]
So, a workaround for this I believe that could be to unlink the `relayout_data` parameter, but I have no idea how that can be done.

As a simpler example, if we create a float input and we print its value, it gets modified after increasing it from the HTML element, which makes sense:

```python
import panel as pn

pn.extension("plotly")

float_input = pn.widgets.FloatInput(name="Number", value=10)

print(float_input.value)
```

However, is there a way to *unlink* `float_input.value` from the JS callback, so it does not get altered when I modify the value through the HTML widget?

@philippjfr If that can be done, I may be able to do the same for a Plotly pane and its `relayout_data`.

**Comment by philippjfr** [comment]
I think this was resolved in https://github.com/holoviz/panel/pull/2445. If not please reopen.

**Comment by julioasotodv** [comment]
Great to know! I will try and test it to confirm if it is solved.

Thank you!

**Comment by CmpCtrl** [comment]
Has there been any update to this? I am finding the same behavior with the versions listed below. 

Also, in my own use case (which i thought was quite similar) the orientation and layout are reset on every change to the `plotly_pane.object` even though i am only updating z values similar to this example.

bokeh                         2.4.2
panel                         0.13.0
param                         1.12.1
plotly                        5.3.1

**Comment by philippjfr** [comment]
This appears to be fixed in 1.0.
